.github/workflows/on.pr.yml

```diff
@@ -0,0 +1,64 @@
+# Guava GitHub CI
+# ---------------------------------------------------------------------------------------------------------------------
+# This is an entrypoint workflow which operates on pull requests; this workflow doesn't do much on its own. Its job is
+# to dispatch `on.build.yml` and check workflows, which can be found in this same directory.
+#
+# PR workflows are slightly different from push workflows (for example, they do not publish snapshots). See the
+# `on.push.yml` workflow. PR and push flows are designed to be invoked separately.
+
+name: PR
+
+on:
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  contents: read
+
+jobs:
+  ## Build the library and provenance material, but don't publish
+  build:
+    name: "Build"
+    uses: ./.github/workflows/ci.build.yml
+    permissions:
+      actions: write
+      contents: write
+      id-token: write
+    with:
+      provenance: true
+      provenance_publish: false
+      snapshot: false
```

Comment:
Entrypoint workflow at `on.pr.yml` runs on `pull_request` events. Build/test steps are invoked by calling into the shared workflows at `ci.build.yml` and `ci.test.yml`.